differ/differ_test.go

```diff
@@ -1451,3 +1453,118 @@ func TestProcessClustersWeeklyDigest(t *testing.T) {
 	zerolog.SetGlobalLevel(zerolog.WarnLevel)
 	kafkaNotifier = originalNotifier
 }
+
+// ---------------------------------------------------------------------------------------
+func TestProduceEntriesToServiceLog(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path != "/rendered_reports" {
+			t.Errorf("Expected to request '/render_reports', got: %s", r.URL.Path)
+		}
+		if r.Header.Get("Content-Type") != "application/json" {
+			t.Errorf("Expected Content-Type: application/json header, got: %s", r.Header.Get("Accept"))
+		}
+		w.WriteHeader(http.StatusOK)
+		_, err := w.Write([]byte(`{"clusters":["first_cluster"],"reports":{"first_cluster":[{"rule_id":"ccx_rules_ocp.external.rules.rule_1","error_key":"RULE_1","resolution":"rule 1 resolution","reason":"This reason is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long.","description":"This reason is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long."},{"rule_id":"ccx_rules_ocp.external.rules.rule_2","error_key":"RULE_2","resolution":"rule 2 resolution","reason":"rule 2 reason","description":"rule 2 error key description"}]}}`))
+		if err != nil {
+			log.Fatal().Msg(err.Error())
+		}
+	}))
+	defer server.Close()
+
+	config := conf.ConfigStruct{
+		ServiceLog: conf.ServiceLogConfiguration{
+			Enabled:            true,
+			TotalRiskThreshold: 1,
+			EventFilter:        "totalRisk > totalRiskThreshold",
+		},
+		Dependencies: conf.DependenciesConfiguration{
+			TemplateRendererServer:   server.URL,
+			TemplateRendererEndpoint: "/rendered_reports",
+			TemplateRendererURL:      server.URL + "/rendered_reports",
+		},
+	}
+	serviceLogEventThresholds.TotalRisk = 1
+	serviceLogEventFilter = "totalRisk > totalRiskThreshold"
+
+	errorKeys := map[string]utypes.RuleErrorKeyContent{
```

Comment:
we have this test data repo that might have what you need in some of the cases (or it might not :))

Reply:
Replaced